src/software/ai/hl/stp/stp.cpp

```diff
@@ -137,17 +145,58 @@ std::vector<std::shared_ptr<Tactic>> STP::assignRobotsToTactics(
     // https://github.com/saebyn/munkres-cpp is the implementation of the Hungarian
     // algorithm that we use here
 
-    if (world.friendlyTeam().numRobots() < tactics.size())
+    auto friendly_team         = world.friendlyTeam();
+    auto& friendly_team_robots = friendly_team.getAllRobots();
+
+    // Special handling for the Goalie, since there should only ever be one and only
+    // one robot per team is ever permitted to be one
+    const std::optional<Robot> goalie = friendly_team.goalie();
+    // Assign the goalie to the first goalie tactic
+    auto it = tactics.begin();
+    if (goalie)
+    {
+        while (it != tactics.end())
+        {
+            if ((*it)->isGoalieTactic())
+            {
+                (*it)->updateRobot(*goalie);
+                break;
+            }
+            it++;
+        }
+    }
+
+    // Discard the goalie tactic we assigned and any other goalie tactics, since
+    // there is only one goalie
+    while (it != tactics.end())
+    {
+        if ((*it)->isGoalieTactic())
+        {
+            // "erase" invalidates the iterator, so we use the returned value
+            it = tactics.erase(it);
+        }
+        else
+        {
+            it++;
+        }
+    }
+
+    std::vector<Robot> non_goalie_robots = friendly_team_robots;
+    if (goalie)
```

Comment:
can we combine this `if(goalie)` with the `if(goalie)` above?

Reply:
Yep, done!